### PR TITLE
Azure - Support blob output for logs/resources

### PR DESCRIPTION
**Pull request by stefangordon**
Works roughly the same way as S3 output with `azure://` prefix.  

Documentation incoming in a separate PR as I have quite a few doc changes to do together.

### CODE REVIEW

File: tools/c7n_azure/c7n_azure/storage_utils.py [hunk 1]
```diff
@@ -86,7 +88,14 @@ def get_storage_keys(storage_account_id):
     def get_storage_from_uri(storage_uri):
         parts = urlparse(storage_uri)
         storage_name = str(parts.netloc).partition('.')[0]
-        container_name = parts.path.partition('/')[2]
+
+        path_parts = os.path.split(parts.path)
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/storage_utils.py]: I think if you were just to do `parts.path.split('/')., You wouldn't need to make all the `strip(...)` calls.

File: tools/c7n_azure/c7n_azure/storage_utils.py [hunk 2]
```diff
@@ -86,7 +88,14 @@ def get_storage_keys(storage_account_id):
...
+        container_name = path_parts[0].strip('/')
+        if len(path_parts) > 1:
+            key_prefix = path_parts[1].strip('/')
```
Comment (aluong) [inline tools/c7n_azure/c7n_azure/storage_utils.py]: If the user provides some blob uri like`https://myaccount.blob.core.windows.net/mycontainer/myblob/seg1/seg2`, would we ignore `seg1` and `seg2`?

Comment (stefangordon) [reply tools/c7n_azure/c7n_azure/storage_utils.py]: At the moment only planning to support a container, and custodian builds the directory structure after the container, but we can probably make it let them specify arbitrary virtual directories

**Comment by stefangordon** [comment]
Ok I believe I covered all feedback, and updated a test to show that we can do arbitrary paths provided by user for blob.

**Comment by stefangordon** [comment]
Having some build issues that don't repro at all locally - will get it investigated.

**Comment by stefangordon** [comment]
closes #2280

**Pull request merged**